pkg/argocd/argocd.go

```diff
@@ -54,29 +55,38 @@ func AddClusterAddFlags(cmd *cobra.Command) (AddClusterCmd, error) {
 	return &addClusterImpl{root, args}, nil
 }
 
-func CheckAppSynced(ctx context.Context, f kube.Factory, ns, name string) (bool, error) {
-	rc, err := f.ToRESTConfig()
-	if err != nil {
-		return false, err
-	}
-
-	c, err := argocdcd.NewForConfig(rc)
-	if err != nil {
-		return false, err
-	}
+func GetAppSyncFn(revision string) kube.WaitFunc {
+	return func(ctx context.Context, f kube.Factory, ns, name string) (bool, error) {
+		rc, err := f.ToRESTConfig()
+		if err != nil {
+			return false, err
+		}
 
-	app, err := c.ArgoprojV1alpha1().Applications(ns).Get(ctx, name, metav1.GetOptions{})
-	if err != nil {
-		se, ok := err.(*errors.StatusError)
-		if !ok || se.ErrStatus.Reason != metav1.StatusReasonNotFound {
+		c, err := argocdcd.NewForConfig(rc)
```

Comment:
argocdcd?

Reply:
yeah, it was supposed t obe argocdcs (for ClientSet)
Do you think I should use a shorter alias?